tc/sgx/trusted_worker_manager/enclave/kme/workload/kme_workload_plug-in.cpp

```diff
@@ -181,6 +192,20 @@ void KMEWorkloadProcessor::Register(
     auto search = sig_key_map.find(unique_id_bytes);
 
     if (search != sig_key_map.end()) {
+
+        // If this is a simulator mode, need not get mr_enclave
+        // from report. Rather read the one recevied from WPE.
+        if (this->isSgxSimulator())
+            mr_enclave_bytes = HexEncodedStringToByteArray(mr_enclave);
+
+        // Compare MRENCLAVE value
+        EnclaveData* enclaveData = EnclaveData::getInstance();
+        ByteArray ext_data = enclaveData->get_extended_data();
+        if (memcmp(ext_data.data(),
+		        mr_enclave_bytes.data(), SGX_HASH_SIZE) != 0) {
+            ThrowIf<ValueError>(true, "WPE MRENCLAVE value didn't match");
```

Comment:
Before throwing error, can you do set status so that in the out data, we can put the reason for registration failure.

Reply:
Done